src/mesh/amr_loadbalance.cpp

```diff
@@ -789,6 +789,7 @@ void Mesh::RedistributeAndRefineMeshBlocks(ParameterInput *pin, ApplicationInput
   RegionSize block_size = GetBlockSize();
 
   BlockList_t new_block_list(nbe - nbs + 1);
+  std::set<LogicalLocation> newly_refined;
```

Comment:
I think we may want this to be a `std::unordered_set` for performance reasons.

Reply:
Agreed, I think, the list could potentially be long and we're checking membership.
What's a good hash function for the unordered_set?  Just morton number?

Speaking of that, I think I populate this list incorrectly, since I only pull from `nbs` to `nbe` of the global block list in populating newly_refined, meaning I need to collate the lists from all ranks.  So, also going to fix that.

Reply:
A hash for `LogicalLocation` is injected into the `std::` namespace, so `std::unordered_map<LogicalLocation>` should just work. 

Good catch. Yeah, I think you need the locations on all ranks.